cmd/livegrep-fetch-reindex/main.go

```diff
@@ -260,7 +283,77 @@ func checkoutOne(r *config.RepoSpec) error {
 	if r.CloneOptions != nil && r.CloneOptions.Depth != 0 {
 		args = append(args, fmt.Sprintf("--depth=%d", r.CloneOptions.Depth))
 	}
-	return callGit("git", args, username, password)
+
+	// Call git-fetch and finish
+	if !(*flagUpdateHead) {
+		_, err = callGit("git", args, username, password, false)
+		return err
+	}
+
+	// We check and update (if needed) the HEAD ref to avoid scenarios where
+	// a remote repo has changed it's head (like a default branch rename/change).
+	// git fetch won't do this, at least not on the mirror clones we use.
+	// See https://public-inbox.org/git/CANWRddPDhM1g6rtu-a2a=EogXD_hOFwSDsgMCbVvB7dibMaEqw@mail.gmail.com/T/#t
+	// for confirmation on this approach from the Git folks.
+	//
+	// To update the HEAD ref we do the following:
+	// 1. Get the remote head ref			- (git ls-remote --symref origin HEAD)
+	// 2. Get the current local head ref	- (git symbolic-ref HEAD)
+	// 3. Compare them. If outdated, update the local to match remote - (git symbolic-ref HEAD new_ref)
+	// We use goroutines to call `git fetch -p` and `git ls-remote --symref origin HEAD` in "parallel"
+	// becase they each take ~1.5s.
+	wg := sync.WaitGroup{}
```

Comment:
Let's use an [`errgroup`](https://pkg.go.dev/golang.org/x/sync/errgroup) here, which will help with the "merging the errors" for us and avoid the error-prone `wg.Add(2)`.

Reply:
Sounds good, this is done now.